clio-integration-test/src/main/python/clio_test.py

```diff
@@ -176,15 +197,57 @@ def expected():
                     'date':    {"type": "string" , "format": "date-time"} }
         response = requests.get(elasticsearch_http_uri + '/read_group/_mapping/default')
         mapping = response.json()['read_group']['mappings']['default']['properties']
+        del mapping['location']
+        properties = { key: schemas[value['type']] for key, value in mapping.items() }
+        result = { 'type': 'object',
+                   'required': [ 'flowcell_barcode', 'lane', 'library_name' ],
+                   'properties': properties }
+        return result
+    response = requests.get(clio_http_uri + '/readgroup/schema/v1')
+    result = response.json()
+    assert result == expected()
+
+
+# I don't know how to derive the required fields.
+#
+def test_json_schema_v2():
```

Comment:
Maybe.  Python makes that kind of ugly if we want to preserve Python2 compatibility.